tee-worker/cli/src/trusted_base_cli/commands/litentry/request_vc.rs

```diff
@@ -509,34 +507,69 @@ impl RequestVcCommand {
 			},
 		};
 
-		let key = Self::random_aes_key();
+		let assertions: Vec<Assertion> = vec![assertion];
+		println!("assertions: {:?}", assertions);
+		let assertions: VecAssertion = BoundedVec::try_from(assertions).unwrap();
+		let alice = get_pair_from_str(trusted_cli, "//Alice", cli);
+		let id: Identity = Identity::from_did(self.did.as_str()).unwrap();
+		println!(">>>id: {:?}", id);
 
-		let top = TrustedCall::request_vc(
+		let (mrenclave, shard) = get_identifiers(trusted_cli, cli);
+		let nonce = get_layer_two_nonce!(alice, cli, trusted_cli);
+		println!(">>>nonce: {}", nonce);
+		let key = Self::random_aes_key();
+		let top = TrustedCall::request_batch_vc(
 			alice.public().into(),
 			id,
-			assertion,
+			assertions,
 			Some(key),
 			Default::default(),
 		)
 		.sign(&KeyPair::Sr25519(Box::new(alice)), nonce, &mrenclave, &shard)
 		.into_trusted_operation(trusted_cli.direct);
 
-		let maybe_vc = if self.stf {
-			perform_trusted_operation::<RequestVCResult>(cli, trusted_cli, &top)
-		} else {
-			perform_direct_operation::<RequestVCResult>(cli, trusted_cli, &top, key)
-		};
-
-		match maybe_vc {
-			Ok(mut vc) => {
-				let decrypted = aes_decrypt(&key, &mut vc.vc_payload).unwrap();
-				let credential_str = String::from_utf8(decrypted).expect("Found invalid UTF-8");
-				println!("----Generated VC-----");
-				println!("{}", credential_str);
-			},
-			Err(e) => {
-				println!("{:?}", e);
-			},
+		let (sender, receiver) = channel::<Result<RequestVcResultOrError, String>>();
+
+		send_direct_batch_vc_request(cli, trusted_cli, &top, key, sender);
+
+		let mut len = 0u8;
+		let mut cnt = 0u8;
+		loop {
```

Comment:
Can we do this inside `send_direct_batch_vc` which should collect responses, and give a vector of responses here and we can just print them. wdyt?